### PR TITLE
Added Husky and nano-staged.

### PR DESCRIPTION
**Pull request by minsoo-web**
I don't use biomejs, so I didn't know if I was formatting correctly or linting. I had minimal defense with recomandation, but I thought it was a negligible barrier (me? I'm not going to install it? lol).

So I created a barrier to make commits safe with husky and nano-staged.

- lint
- formatting
- type checking

### CODE REVIEW

**Comment by manudeli** [comment]
@minsoo-web Could you resolve conflict? I want to merge this but I can't because of conflict

File: biome.jsonc [hunk 1]
```diff
@@ -7,16 +7,22 @@
     "enabled": true,
     "rules": {
       "recommended": true
-    }
+    },
+    "ignore": ["node_modules", "dist", ".next"]
```
Comment (2-NOW) [inline biome.jsonc:11]: Thanks for pointing out the good stuff :D

File: biome.jsonc [hunk 2]
```diff
@@ -7,16 +7,22 @@
...
   },
   "json": {
+    "parser": {
+      "allowTrailingCommas": true
+    },
     "formatter": {
-      "enabled": true
+      "enabled": true,
+      "trailingCommas": "none"
```
Comment (2-NOW) [inline biome.jsonc:25]: default is "none", so I don't think it's necessary to add it.

ref: https://biomejs.dev/reference/configuration/#jsonformattertrailingcommas

File: biome.jsonc [hunk 3]
```diff
@@ -7,16 +7,22 @@
...
   "formatter": {
     "enabled": true,
     "indentWidth": 2,
-    "indentStyle": "space"
+    "indentStyle": "space",
+    "ignore": ["node_modules", "dist", ".next"]
   },
   "json": {
+    "parser": {
+      "allowTrailingCommas": true
```
Comment (2-NOW) [inline biome.jsonc:21]: Thanks for pointing out the good stuff :D

File: biome.jsonc (hunk 2 above)
Comment (minsoo-web) [reply biome.jsonc]: Thanks!!! I also felt this was unnecessary and fixed it!
[fix: delete useless config](https://github.com/coauthors/coauthors/pull/5/commits/e85ed745d6e47dc3d71874f87674db6a96cca82f)

File: biome.jsonc (hunk 3 above)
Comment (minsoo-web) [reply biome.jsonc]: This seems like an unnecessary setting, so I deleted it!
[fix: delete useless config](https://github.com/coauthors/coauthors/pull/5/commits/e85ed745d6e47dc3d71874f87674db6a96cca82f)

Comment (2-NOW) [reply biome.jsonc]: No. This setting allows you to parse trailing commas in json. If you don't have this setting, the last comma in json will not be formatted when formatting. But if you set this option, the trailing comma in json will be formatted. Try to test it :D

Comment (minsoo-web) [reply biome.jsonc:21]: Crazy foolishness restore

[chore: restore allowTrailingCommas options](https://github.com/coauthors/coauthors/pull/5/commits/7b842b7259d2a28b309f8716e7a16c836f6412d3)

**Review by manudeli — APPROVED** [review]
👍

**Pull request merged**